mail/src/main/kotlin/com/ritense/mail/domain/filters/BlacklistFilter.kt

```diff
@@ -41,17 +42,23 @@ class BlacklistFilter(
 ) : MailFilter {
 
     override fun doFilter(rawMailMessage: RawMailMessage): Optional<RawMailMessage> {
-        rawMailMessage
-            .recipients
-            .filterBy { !blacklistService.isBlacklisted(it.email.get()) }
-        return Optional.of(rawMailMessage)
+        return doFilterInternal(rawMailMessage)
     }
 
     override fun doFilter(templatedMailMessage: TemplatedMailMessage): Optional<TemplatedMailMessage> {
-        templatedMailMessage
+        return doFilterInternal(templatedMailMessage)
+    }
+
+    private fun <T: HasRecipients> doFilterInternal(mailMessage: T): Optional<T> {
+        mailMessage
             .recipients
             .filterBy { !blacklistService.isBlacklisted(it.email.get()) }
-        return Optional.of(templatedMailMessage)
+
+        return if (mailMessage.recipients.isPresent) {
+            Optional.of(mailMessage)
+        } else {
+            Optional.empty()
```

Comment:
Do we want to add debug logging so users might understand why their mail is not being sent?

Reply:
fixed